### PR TITLE
Add support for creation of Docker containers for platform services

### PR DESCRIPTION
**Pull request by pazaan**
Add Dockerfile per service that needs to run.
Update `Makefile` to support Alpine Linux's lack of `xargs -L`

### CODE REVIEW

**Comment by pazaan** [comment]
[Docker for Development](https://trello.com/c/iXu5Fuvj/176-docker-for-development)

**Comment by darinkrauss** [comment]
Do we need a `.dockerignore` file to ignore `.git` and `.gitignore`?

**Comment by pazaan** [comment]
The `build` process requires a git workspace to exist. I also don't remove the .git and .gitignore from the fs layer after the container build so that we can mount local code to the container and still successfully `build`. A production-deployable container would work differently, and be trimmed down a lot more.

File: Makefile [hunk 1]
```diff
@@ -112,8 +112,8 @@ build-list:
 
 build: check-environment
 	@echo "go build"
-	@cd $(ROOT_DIRECTORY) && $(FIND_MAIN_CMD) | $(TRANSFORM_MKDIR_CMD) | xargs -L1 $(MKDIR_CMD)
-	@cd $(ROOT_DIRECTORY) && $(FIND_MAIN_CMD) | $(TRANSFORM_GO_BUILD_CMD) | xargs -L1 $(GO_BUILD_CMD)
+	@cd $(ROOT_DIRECTORY) && $(FIND_MAIN_CMD) | $(TRANSFORM_MKDIR_CMD) | while read line; do $(MKDIR_CMD) $$line; done
```
Comment (pazaan) [inline Makefile]: Yes, it works in macOS and in the Vagrant box (Ubuntu).
Good point about consistency - sorry!

File: Makefile [hunk 2]
```diff
@@ -112,8 +112,8 @@ build-list:
 
 build: check-environment
 	@echo "go build"
-	@cd $(ROOT_DIRECTORY) && $(FIND_MAIN_CMD) | $(TRANSFORM_MKDIR_CMD) | while read line; do $(MKDIR_CMD) $$line; done
-	@cd $(ROOT_DIRECTORY) && $(FIND_MAIN_CMD) | $(TRANSFORM_GO_BUILD_CMD) | while read line; do $(GO_BUILD_CMD) $$line; done
+	@cd $(ROOT_DIRECTORY) && $(FIND_MAIN_CMD) | $(TRANSFORM_MKDIR_CMD) | while read LINE; do $(MKDIR_CMD) $${LINE}; done
+	@cd $(ROOT_DIRECTORY) && $(FIND_MAIN_CMD) | $(TRANSFORM_GO_BUILD_CMD) | while read LINE; do $(GO_BUILD_CMD) $${LINE}; done
```
Comment (darinkrauss) [inline Makefile]: Is it curly braces or parentheses in a Makefile?

Comment (pazaan) [inline Makefile]: Parentheses for a Makefile variable, curly braces for a shell variable.

**Pull request merged**